### ISSUE TITLE
Astro Islands fails with rolling update

### ISSUE THREAD

**Issue by sasoria** [withastro/astro#11851]
### Astro Info

```block
Astro                    v4.14.0
Node                     v20.13.1
System                   macOS (arm64)
Package Manager          npm
Output                   server
Adapter                  @astrojs/node
Integrations             @astrojs/react
                         importmap
```

### Describe the Bug

I've experimented with Astro Islands in our dev cluster and it looks like the fetch to get the Server Island fails with status code 500 when the application is being deployed. We use a [rolling update](https://kubernetes.io/docs/tutorials/kubernetes-basics/update/update-intro/) on Kubernetes which is quite common.

The problem as I presume it, is that the new application that is being deployed exists at the same time as the old one. Hence, the html from an old version of the application can try to fetch the Server Island from a new one. 

Perhaps something about the encryption is causing this to fail? The error I see on the server is this one:
```
[ERROR] OperationError: The operation failed for an operation-specific reason
```

Screenshots
---
![Screenshot 2024-08-27 at 15 04 10](https://github.com/user-attachments/assets/f2a92337-1bf5-4b1d-b910-f528c0b86d8c)

![Screenshot 2024-08-27 at 15 02 24](https://github.com/user-attachments/assets/cdd9e295-3d85-4917-9c48-fb52b8549157)

Is there anything that can be done to mitigate this? Let me know if there is any other information you want me to provide.

### What's the expected result?

To fetch the Server Island correctly when deploying with a rolling update.

### Link to Minimal Reproducible Example

none

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by matthewp** [comment]
Yes, your frontend is deploying more slowly than your backend (or vice-versa) so the keys don't match. Likely what we'd want to do is provide the key to you at build time and allow you to re-use the same key on every deployment as an environment variable. Would that be possible?

**Comment by matthewp** [comment]
@sasoria Actually the key would likely need to be set in your *build server* rather than the production server. Does that change anything?

**Comment by sasoria** [comment]
I can't see how that would change things as long as there is a way to provide the key at build time. I think it's worth a shot!

**Comment by matthewp** [comment]
@sasoria can you try with `astro@experimental--crypto-env`?

**Comment by sasoria** [comment]
Thank you, I'm testing it now and will provide the result to the PR.

**Comment by matthewp** [comment]
This is fixed in `astro@4.15.4`. If you look at the release notes you generate a key with `astro create-key`. Thanks for raising this!
